### PR TITLE
2022 14 db snp column

### PR DESCRIPTION
**Pull request by Swaathik**
## Purpose
Display dbSNP cross references data(from FeatureCrossReferences unlike other xrefs). [JIRA](https://www.ebi.ac.uk/panda/jira/browse/TRM-28089)

## Approach
Information displayed are as follows:

- Variant change with link to UniProt variant page
- rsID
- Links to dbSNP and Ensembl

Description is there in the data but left out as it started to look cluttered. I will attach screenshot for reference.
Example in Legacy for one such with extra description - https://legacy.uniprot.org/uniprot/?query=database%3A%28type%3Adbsnp%29+AND+A0A1B0GTW7&sort=score

Links to dbSNP and UniProt variant pages are added in Disease and variants section following Elisabeth's comment in the JIRA.(http://localhost:8080/uniprotkb/P78363/entry#disease_variants)

## Testing
Updated the snapshots

## Checklist
- [x] My PR is scoped properly, and “does one thing only”
- [x] I have reviewed my own code
- [x] I have checked that linting checks pass and type safety is respected
- [x] I have checked that tests pass and coverage has at least improved, and if not explained the reasons why

### CODE REVIEW

File: src/uniprotkb/components/protein-data-views/DiseaseInvolvementView.tsx [hunk 1]
```diff
@@ -15,6 +15,21 @@ import { Namespace } from '../../../shared/types/namespaces';
 import { FeatureDatum } from './UniProtKBFeaturesView';
 
 import styles from './styles/variation-view.module.scss';
+import externalUrls from '../../../shared/config/externalUrls';
```
Comment (aurel-l) [inline src/uniprotkb/components/protein-data-views/DiseaseInvolvementView.tsx:18]: Just a small note on the order of imports, this should be higher

File: src/uniprotkb/components/protein-data-views/DiseaseInvolvementView.tsx [hunk 2]
```diff
@@ -15,6 +15,21 @@ import { Namespace } from '../../../shared/types/namespaces';
...
+    <ExternalLink url={externalUrls.UniProt(variant.featureId || '')} noIcon>
+      {variant.alternativeSequence?.originalSequence || <em>missing</em>}
+      {'>'}
+      {variant.alternativeSequence?.alternativeSequences?.[0] || (
+        <em>missing</em>
+      )}
+    </ExternalLink>
+  ) : (
+    <em>missing</em>
+  );
```
Comment (aurel-l) [inline src/uniprotkb/components/protein-data-views/DiseaseInvolvementView.tsx:32]: I think all of them should be wrapped with the external link component, even if they are "missing". In the page example you gave, the first displayed variant had "missing" written but still had a featureId in the data, so it should still link the the corresponding page.

Comment (Swaathik) [reply src/uniprotkb/components/protein-data-views/DiseaseInvolvementView.tsx:32]: But it takes to the expasy home page. I assume that doesn't exist. 
this has a change - -https://web.expasy.org/variant_pages/VAR_084833.html
while this has 'missing' - https://web.expasy.org/variant_pages/VAR_012494.html

Comment (Swaathik) [reply src/uniprotkb/components/protein-data-views/DiseaseInvolvementView.tsx:32]: I checked how it is done in legacy. It is not linked there either

Comment (aurel-l) [reply src/uniprotkb/components/protein-data-views/DiseaseInvolvementView.tsx:32]: Ah, right. I just saw that the ID is there in the data, but didn't realise the corresponding link in the expasy site was leading nowhere.
In which case everything is fine 👌

**Review by aurel-l — APPROVED** [review]

**Pull request merged**